lib/screens/component_screen.dart

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+const List<NavigationDestination> appBarDestinations = [
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.home_outlined),
+    label: 'About Us',
+    selectedIcon: Icon(Icons.home),
+  ),
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.star_border_outlined),
+    label: 'Projects',
+    selectedIcon: Icon(Icons.star),
+  ),
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.event_outlined),
+    label: 'Events',
+    selectedIcon: Icon(Icons.event),
+  ),
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.rss_feed_outlined),
+    label: 'Blog',
+    selectedIcon: Icon(Icons.rss_feed),
+  ),
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.contact_page_outlined),
+    label: 'Contact us',
+    selectedIcon: Icon(Icons.contact_page),
+  ),
+];
+
+final List<NavigationRailDestination> navRailDestinations = appBarDestinations
+    .map(
+      (destination) => NavigationRailDestination(
+        icon: Tooltip(
+          message: destination.label,
+          child: destination.icon,
+        ),
+        selectedIcon: Tooltip(
+          message: destination.label,
+          child: destination.selectedIcon,
+        ),
+        label: Text(destination.label),
+      ),
+    )
+    .toList();
+
+const List<Widget> exampleBarDestinations = [
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.explore_outlined),
+    label: 'Explore',
+    selectedIcon: Icon(Icons.explore),
+  ),
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.pets_outlined),
+    label: 'Pets',
+    selectedIcon: Icon(Icons.pets),
+  ),
+  NavigationDestination(
+    tooltip: "",
+    icon: Icon(Icons.account_box_outlined),
+    label: 'Account',
+    selectedIcon: Icon(Icons.account_box),
+  )
+];
+
+class NavigationBars extends StatefulWidget {
+  final void Function(int)? onSelectItem;
+  final int selectedIndex;
+  final bool isExampleBar;
+
+  const NavigationBars(
+      {super.key,
+      this.onSelectItem,
+      required this.selectedIndex,
+      required this.isExampleBar});
+
+  @override
+  State<NavigationBars> createState() => _NavigationBarsState();
+}
+
+class _NavigationBarsState extends State<NavigationBars> {
+  int _selectedIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedIndex = widget.selectedIndex;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return NavigationBar(
```

Comment:
Is this a custom NavigationBar? The implementation seems very close to the BottomNavigationBar, anyway looks good! 

I'd recommend looking at AutoRoute https://pub.dev/packages/auto_route for this! It's a little tricky to learn, but once you do it's so good!